### PR TITLE
[FEATURE] Android desugaring warning

### PR DESCRIPTION
**Pull request by roliveiravictor**
#4137

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,8 +52,11 @@ We would love to hear from the larger community: please provide feedback proacti
 
 ## Requirements
 
-Unless otherwise noted, all published artifacts support Java 8 or higher. See [CONTRIBUTING.md](./CONTRIBUTING.md)
-for additional instructions for building this project for development.
+Unless otherwise noted, all published artifacts support Java 8 or higher.
+
+**Android Disclaimer:** _It's highly recommended enabling [desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring) in order to avoid odd Runtime exceptions (i.e. Issue [#4137](https://github.com/open-telemetry/opentelemetry-java/issues/4137))._
```
Comment (breedx-splk) [inline README.md:57]: I think it should go beyond a recommendation and just say requirement:

```suggestion
**Android Disclaimer:** For compatibility reasons, [library desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring) must be enabled. (i.e. Issue [#4137](https://github.com/open-telemetry/opentelemetry-java/issues/4137)).
```

Comment (jkwatson) [reply README.md:57]: There's only one exception...if you aren't targeting API levels that require it, then it wouldn't be needed.

**Review by mateuszrzeszutek — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ We would love to hear from the larger community: please provide feedback proacti
 
 Unless otherwise noted, all published artifacts support Java 8 or higher.
 
-**Android Disclaimer:** _It's highly recommended enabling [desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring) in order to avoid odd Runtime exceptions (i.e. Issue [#4137](https://github.com/open-telemetry/opentelemetry-java/issues/4137))._
+**Android Disclaimer:** For compatibility reasons, [library desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring) must be enabled. (i.e. Issue [#4137](https://github.com/open-telemetry/opentelemetry-java/issues/4137)).
```
Comment (jkwatson) [inline README.md:57]: I'm not sure referencing an issue in a README is a great idea. I'd prefer we just stick with the note, and not reference the issue, personally.

**Review by jkwatson — APPROVED** [review]
Good addition. thanks!

**Review by jack-berg — APPROVED** [review]

**Pull request merged**